File: AtlasWeb/Core/wwwroot/app/Fw/Models/AtlasWebSocketContainer.ts

export enum AtlasSourceType {
    System = 1,
    User = 2,
}
export enum AtlasContentType {
    Message = 1,
    Notification = 2,
    Action = 3,
    Emergency = 4,
    LaboratoryPanicNotification = 5,
}


export class AtlasWebSocketContainer {
    public users: Array<string> = new Array<string>();
    public actionData: any;
    public content: any;
    public contentType: AtlasContentType ;
    public sourceType: AtlasSourceType;
    public dataProcessedSuccessfully: boolean;
    public status: string ;
    public headerDefinition: string;
    public createTimeStr: string;
    public createTime: Date;
    public notificationID : string;

    constructor(){
        this.createTime = new Date();
        this.createTimeStr =  this.createTime.toLocaleTimeString();
    }

}

File: AtlasWeb/Core/AtlasWebSocketManager/Containers/AtlasWebSocketContainer.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Core.AtlasWebSocketManager.Containers
{
    interface I_AtlasWebSocketContainer
    {
        List<string> users { get; set; }
        object actionData { get; set; }
        string content { get; set; }
        AtlasContentType contentType { get; set; }
        AtlasSourceType sourceType { get; set; }
        bool dataProcessedSuccessfully { get; set; }
        string status { get; set; }
        string headerDefinition { get; set; }
        string currentTimeStr { get; set; }
        DateTime currentTime { get; set; }
    }

    public class AtlasWebSocketContainer : I_AtlasWebSocketContainer
    {
        public List<string> users { get; set; }
        public object actionData { get; set; }
        public string content { get; set; }
        public AtlasContentType contentType { get; set; }
        public AtlasSourceType sourceType { get; set; }
        public string status { get; set; }
        public bool dataProcessedSuccessfully { get; set; }

        //Notification un eventin turunu belirleyen bir isim (orn : YeniHastaKaydi)
        public string headerDefinition { get; set; }

        public string currentTimeStr { get; set; }
        public DateTime currentTime { get; set; }

        public AtlasWebSocketContainer()
        {
            this.currentTime = DateTime.Now;
            this.currentTimeStr = this.currentTime.ToShortTimeString();
        }
    }


}
